packages/cli/src/config/loadStoreConfig.ts

```diff
@@ -1,142 +1,100 @@
-import path from "path";
-import { SchemaType } from "@latticexyz/schema-type";
-import { defaultKeyTuple, defaultStoreImportPath, defaultTablePath } from "../utils/constants.js";
-import { ERRORS, MUDError } from "../utils/errors.js";
+import { SchemaType, getStaticByteLength } from "@latticexyz/schema-type";
+import { z, ZodError } from "zod";
+import { fromZodErrorCustom } from "../utils/errors.js";
+import { BaseRoute, ObjectName, OrdinaryRoute, ValueName } from "./commonSchemas.js";
 import { loadConfig } from "./loadConfig.js";
-import { isArray, isRecord, isString } from "./utils.js";
-
-// Based on hardhat's config (MIT)
-// https://github.com/NomicFoundation/hardhat/tree/main/packages/hardhat-core
 
+const TableName = ObjectName;
+const KeyName = ValueName;
+const ColumnName = ValueName;
+
+const PrimaryKey = z
+  .nativeEnum(SchemaType)
+  .refine((arg) => getStaticByteLength(arg) > 0, "Primary key must not use dynamic SchemaType");
+const PrimaryKeys = z.record(KeyName, PrimaryKey).default({ key: SchemaType.BYTES32 });
+
+const Schema = z
+  .record(ColumnName, z.nativeEnum(SchemaType))
+  .refine((arg) => Object.keys(arg).length > 0, "Table schema may not be empty");
+
+const FullTable = z
+  .object({
+    route: OrdinaryRoute.default("/tables"),
```

Comment:
looks like this can't be empty correct? I think we should allow registering tables at the root level of a World (eg. `/RootTable`), which would require passing the empty root route here, however this wouldn't work as nicely with using the same route as output path for the generated file

Reply:
Correct, it's the one case that messes up route->folder.
Do we really need root tables?

Reply:
I think it might be useful, if only to allow users to minimize route lengths. Just feels like we shouldn't prevent a setup that uses tables at the root. We could even just put the auto-generated file into the "base directory" (right now foundry's `src` config - but we could add a config field to allow overriding the table base directory).

Not blocking for this PR though.

Reply:
Just allowing root routes is easy actually - replace validateRoute with validateBaseRoute in zod schema.
Having an extra config option for baseDirectory is also pretty trivial, although baseRoute+baseDirectory feels like a bit much.

Btw I don't bundle systems and tables together here. Systems are writers, and for systems that have access to everything (i.e. use delegatecall) I agree root is the best place.

Problems I have with root tables:
- Encourages flat setup. It's intuitive for small projects, but projects that need MUD aren't really small imo. No root simplifies the switch from "/tables" to "/skills", "/items" etc.
- Tables no longer have access groups. If you want to give access to several at once, you'd give root route. You shouldn't give out root access.
- Unlike systems, root tables don't signify a root permission. They don't *have* to exist

Rather than root routes I think it'd be more useful to consider shorthands/config reorgs for setting the same route for multiple tables/systems

My current folders (they would more or less become routes in v2, and this sorta setup is what I'm optimizing for):
![Screenshot 2023-02-28 033836](https://user-images.githubusercontent.com/62725447/221722520-79e8d857-74de-4577-8d6e-0b61da7c31d8.png)

Reply:
On the other hand, if route and folder path are completely separated, it feels like my problems with root tables kinda evaporate. And maybe we should have `route: ""` by default, but `path: "/tables"` by default, and have no significant relationship between the 2

Reply:
> Encourages flat setup. It's intuitive for small projects, but projects that need MUD aren't really small imo. No root simplifies the switch from "/tables" to "/skills", "/items" etc.

I think with v2 there are a lot of good reasons for small projects to use MUD too (eg default compatibility with indexers and network stack), so my preference would be to make all project configurations possible, including simple/flat ones.

I still agree with making more complex / nested project structures like your example easy to achieve too. After thinking about it more I think separating on-chan `route` and filesystem `outputPath` like you proposed makes the most sense, because it allows for any configuration of routes and filesystem structure. This also makes the id computation consistent with systems, where it's always `keccak256(baseRoute/route/SystemName)` with `route` being `""` by default. (Also for systems `route` is independent from the file system path too because they are not auto-generated/)

Reply:
While working on the system deploy script I am realizing it would be most intuitive if `route` allows users to completely override the route, without a default `/SystemName` or `/TableName` appended at the end. So basically if `route` is omitted the default route should be `baseRoute/SystemOrTableName`, while if the route key is present it should be `baseRoute/route`. Otherwise we might prevent some configurations where the route has to be some explicit value that's not the contract name.

Reply:
> While working on the system deploy script I am realizing it would be most intuitive if `route` allows users to completely override the route, without a default `/SystemName` or `/TableName` appended at the end. So basically if `route` is omitted the default route should be `baseRoute/SystemOrTableName`, while if the route key is present it should be `baseRoute/route`. Otherwise we might prevent some configurations where the route has to be some explicit value that's not the contract name.

Sounds good! I can change that in store config in one of the upcoming PRs